src/ray/core_worker/core_worker.cc

```diff
@@ -1390,6 +1402,33 @@ void CoreWorker::HandleWaitForRefRemoved(const rpc::WaitForRefRemovedRequest &re
                                             owner_address, ref_removed_callback);
 }
 
+void CoreWorker::TryKillTask(const TaskID &task_id, int num_tries) {
+  if (num_tries == 0) {
+    RAY_LOG(WARNING) << "Failed to Kill Task: " << task_id;
+    return;
+  }
+  {
+    absl::MutexLock lock(&mutex_);
+    if (main_thread_task_id_ == task_id) {
+      RAY_LOG(INFO) << "Interrupting main: " << main_thread_task_id_;
+      kill_main_thread_();
+      return;
+    }
+  }
+  kill_retry_timer_.expires_from_now(boost::asio::chrono::milliseconds(50));
+  kill_retry_timer_.async_wait(
+      boost::bind(&CoreWorker::TryKillTask, this, task_id, num_tries - 1));
+}
+
+void CoreWorker::HandleKillTask(const rpc::KillTaskRequest &request,
```

Comment:
what happen if the task is inside worker's scheduling queue and timeout expired?

Reply:
For direct tasks, the task is never entered into the scheduling queue, it is immediately executed.

Reply:
What about for actor tasks? Is that not supported at all?

Reply:
Correct.